gobblin-core/src/main/java/gobblin/source/extractor/watermark/SimpleWatermark.java

```diff
@@ -67,11 +67,11 @@ public int getDeltaNumForNextWatermark() {
     long startNum = lowWatermarkValue;
     long endNum = highWatermarkValue;
     boolean longOverflow = false;
-    while (startNum <= endNum && !longOverflow) {
+    while (startNum < endNum && !longOverflow) {
       longOverflow = (Long.MAX_VALUE - interval < startNum);
-      nextNum = longOverflow ? Long.MAX_VALUE : startNum + interval;
-      intervalMap.put(startNum, (nextNum <= endNum ? nextNum : endNum));
-      startNum = nextNum + deltaForNextWatermark;
+      nextNum = longOverflow ? Long.MAX_VALUE : Math.min(startNum + interval, endNum);
```

Comment:
This is not the same logic in the overflow case. The old logic would put endNum instead of Long.MAX_VALUE for overflow case.